public/script.js

```diff
@@ -1,23 +1,62 @@
 // Find the button
 // when you click the button, it should call a function "getSnippet"
+const inputs = document.getElementById('paragraphInput');
 const paragraph = document.getElementById('text-snippet');
-
 const button = document.getElementById('generate-snippet-btn');
 button.addEventListener('click', getSnippet);
 
 function getSnippet() {
     // call the server using "fetch" to '/generateText' API route
     // wait for it to return response (read about **promises** and **async await**), take the text that it provided
     // find the "text-snippet" element and set it's text to the response's text
+    const sumNum = inputs.value
+    const sumNum1 = {score: inputs.value}
     let speech = new SpeechSynthesisUtterance();
     speech.lang = "en";
 
-    fetch('/generateText')
+    // detects number && letters
+    let letters = /^[A-Za-z]+$/;
+    let numbers = /^[0-9]+$/;
+    
+    const options = {
+      method: "POST",
+      body: JSON.stringify(sumNum1) ,
+      headers: {"Content-type": "application/json; charset=UTF-8"}
+      
+    }
+    console.log(sumNum1)
+
+    // checks for letters numbers and empty space.
+    if(sumNum === ""){
+        setErrorFor( inputs, 'please enter a number')
+    }
+    else if (sumNum.match(letters)){
+        setErrorFor( inputs, 'please enter a number')
+    }
```

Comment:
If you remove these if statements, will it still work?
Because in your next if, you're checking if the the entered text are numbers.
So if you enter some random text that aren't numbers, it's not gonna enter that if statement. So you could add an `else` after that and call the `setErrorFor` function in it.

Shrinking down the code is always good